### PR TITLE
Bug 1746172: Fix bug with counting of volumes by matching provisioner names

### PR DESCRIPTION
**Pull request by gnufied**
Fix counting of volumes by matching provisioner names.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1746172

### CODE REVIEW

**Comment by bertinatto** [comment]
/assign @mfojtik 

for approval

File: vendor/k8s.io/kubernetes/pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -472,6 +484,20 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []v1.Volume, namespace s
 	return nil
 }
 
+// matchProvisioner helps identify if the given PVC belongs to the running predicate.
+func (c *MaxPDVolumeCountChecker) matchProvisioner(pvc *v1.PersistentVolumeClaim) bool {
+	if pvc.Spec.StorageClassName == nil {
+		return false
+	}
+
+	storageClass, err := c.scInfo.GetStorageClassInfo(*pvc.Spec.StorageClassName)
```
Comment (smarterclayton) [inline vendor/k8s.io/kubernetes/pkg/scheduler/algorithm/predicates/predicates.go]: Is this a significant performance impact in this code path?

Comment (bertinatto) [reply vendor/k8s.io/kubernetes/pkg/scheduler/algorithm/predicates/predicates.go]: I ran a few benchmarks when I was working on this (https://github.com/kubernetes/kubernetes/pull/80360), but I didn't noticed any significant performance impact. Unfortunately I didn't save the numbers.

**Pull request merged**